ddtrace/internal/packages.py

```diff
@@ -0,0 +1,32 @@
+import typing
+
+
+Distribution = typing.NamedTuple("Distribution", [("name", str), ("version", str)])
+
+_DISTRIBUTIONS = None
+
+
+def get_distributions():
+    # type: () -> typing.List[Distribution]
+    global _DISTRIBUTIONS
+    if _DISTRIBUTIONS is not None:
+        return _DISTRIBUTIONS
+
+    try:
+        import importlib.metadata as importlib_metadata
+    except ImportError:
+        import importlib_metadata  # type: ignore[no-redef]
+
+    pkgs = []
+    pkg_names = set()
+    for dist in importlib_metadata.distributions():
+        metadata = dist.metadata
+        name = metadata["name"]
+        version = metadata["version"]
+        if name and version and name not in pkg_names:
```

Comment:
are there cases when we don't have a name or a version?

are there cases when we get duplicate distribution names back from importlib? do we have to worry about deduplicating here?


if we don't have to worry about deduplicating, we could just store the distributions as a `set()` since a typing.NamedTuple is hashable. e.g.

```python
>>> Distribution = typing.NamedTuple("Distribution", [("name", str), ("version", str)])
>>> dists = set()
>>> dists.add(Distribution("name", "version"))
>>> dists.add(Distribution("name", "version"))
>>> dists.add(Distribution("name", "version"))
>>> dists.add(Distribution("name", "version"))
>>> dists
{Distribution(name='name', version='value')}
```

I am not sure if that is actually what we want, and would only work if know we won't get duplicate names, because the set approach will only dedupe if we have the same name/version combination.

Reply:
In the get distribution there were cases where duplicate packages were returned by `importlib_metadata.distributions()`. That's why I added the pkg_name set to filter out duplicates. 

I'll just add `Distribution` to a set since that's a lot cleaner. I haven't seen a case were name or version is None but it can occur if a metadata file is malformed. It's doesn't hurt to add a check here.